etl/steps/export/github/co2_data/latest/owid_co2.py

```diff
@@ -0,0 +1,363 @@
+
+    # Load primary energy consumption dataset (by different sources in our 'energy' namespace).
+    ds_energy = paths.load_dataset("primary_energy_consumption")
+
+    # Load population dataset.
+    ds_population = paths.load_dataset("population")
+
+    # Load countries-regions dataset (required to get ISO codes).
+    ds_regions = paths.load_dataset("regions")
+
+    # Gather all required tables from all datasets.
+    tb_gcp = ds_gcp["global_carbon_budget"]
+    tb_jones = ds_jones["national_contributions"]
+    tb_climate_watch_ghg = ds_climate_watch["greenhouse_gas_emissions_by_sector"]
+    tb_climate_watch_ch4 = ds_climate_watch["methane_emissions_by_sector"]
+    tb_climate_watch_n2o = ds_climate_watch["nitrous_oxide_emissions_by_sector"]
+    tb_energy = ds_energy["primary_energy_consumption"]
+    tb_gdp = ds_gdp["maddison_project_database"]
+    tb_population = ds_population["population"]
+    tb_regions = ds_regions["regions"]
+
+    #
+    # Process data.
+    #
+    # Choose required columns and rename them.
+    tb_gcp = tb_gcp.reset_index()[list(GCP_COLUMNS)].rename(columns=GCP_COLUMNS, errors="raise")
+    tb_jones = tb_jones.reset_index()[list(JONES_COLUMNS)].rename(columns=JONES_COLUMNS, errors="raise")
+    tb_climate_watch_ghg = tb_climate_watch_ghg.reset_index()[list(CLIMATE_WATCH_GHG_COLUMNS)].rename(
+        columns=CLIMATE_WATCH_GHG_COLUMNS, errors="raise"
+    )
+    tb_climate_watch_ch4 = tb_climate_watch_ch4.reset_index()[list(CLIMATE_WATCH_CH4_COLUMNS)].rename(
+        columns=CLIMATE_WATCH_CH4_COLUMNS, errors="raise"
+    )
+    tb_climate_watch_n2o = tb_climate_watch_n2o.reset_index()[list(CLIMATE_WATCH_N2O_COLUMNS)].rename(
+        columns=CLIMATE_WATCH_N2O_COLUMNS, errors="raise"
+    )
+    tb_energy = tb_energy.reset_index()[list(PRIMARY_ENERGY_COLUMNS)].rename(
+        columns=PRIMARY_ENERGY_COLUMNS, errors="raise"
+    )
+    tb_gdp = tb_gdp.reset_index()[list(GDP_COLUMNS)].rename(columns=GDP_COLUMNS, errors="raise")
+    tb_population = tb_population.reset_index()[list(POPULATION_COLUMNS)].rename(
+        columns=POPULATION_COLUMNS, errors="raise"
+    )
+    tb_regions = tb_regions.reset_index()[list(REGIONS_COLUMNS)].rename(columns=REGIONS_COLUMNS, errors="raise")
+
+    # Combine tables.
+    combined = combine_tables(
+        tb_gcp=tb_gcp,
+        tb_jones=tb_jones,
+        tb_climate_watch_ghg=tb_climate_watch_ghg,
+        tb_climate_watch_ch4=tb_climate_watch_ch4,
+        tb_climate_watch_n2o=tb_climate_watch_n2o,
+        tb_energy=tb_energy,
+        tb_gdp=tb_gdp,
+        tb_population=tb_population,
+        tb_regions=tb_regions,
+    )
+
+    # Prepare outputs.
+    combined = prepare_outputs(combined=combined, ds_regions=ds_regions)
+
+    gh.commit_file_to_github(
+        combined.to_csv(),
+        repo_name="co2-data",
+        file_path="owid-co2-data.csv",
+        commit_message=":bar_chart: Automated update",
+        branch="master",
```

Comment:
I'm not sure about this. Would it be possible to create a PR instead (maybe some of what we do for `draft-pr`)? If that's too complicated, no problem.
Usually, when updating CO2 or energy data, I also manually update the README a bit, to explain the changes and update the versions. Ideally, those README changes could also happen here. But then we would need to commit multiple files.

Reply:
One option would be to have an `etl.helper` function to handle whether you want to push directly or create a PR, and which files you want to commit.

Reply:
Oh right, I didn't realize that CO2 and monkeypox would have different update frequency and automatic updates might not be feasible. Creating a PR would definitely be an option, but it'd be better to do it during the next CO2 update to get a better sense of what it entails.

How about this workflow for now:
1. You create a PR with `etl d pr my-update ...`
2. In the export step, we only commit to a branch if an env variable is set. E.g.
```
if os.environ["CO2_BRANCH"]:
   gh.commit_file_to_github(..., branch=os.environ["CO2_BRANCH"])
```
3. You call it with `CO2_BRANCH=my-update etlr ... --export`

You can iterate on this, make sure your PR looks good, and then merge.

For monkeypox, it'd be committing right to master if `ETL_ACTIONS_COMMIT` env is set (in production).

Reply:
Thanks. It's not ideal to have an env variable for each repos (one for co2, one for energy, one eventually for poverty, maybe others). But for now if it works, I'm fine with it.